Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,144 @@
+===============================
+Virtual Bus Devices and Drivers
+===============================
+
+See <linux/virtual_bus.h> for the models for virtbus_device and virtbus_driver.
+
+This bus is meant to be a minimalist software-based bus used for splitting the
+functionality of a PCI device into orthogonal parts. Each device attached
+to the bus represents an individual function of its parent PCI device. The
+virtual bus can be used by multiple subsystems and there is no implicit
+relationship between the devices attached to it. Only the devices created by
+the same PCI device share a common parent and establish a shared communication
+channel with the parent. The virtual bus has some similarities with MFD but MFD
+is not suitable for PCI devices because it utilizes the platform bus that is
```

Comment:
@ranj063 Maybe clarify this refers to Linux MFD framework. PCI devices can also host multiple function and PCI multifunction devices are a different thing. This is about exposing a single PCI function (the audio DSP) in multiple Linux subsystems.

Reply:
I like the verbiage used here by kv2019i if its also true for sound use-case.
"This is about exposing a single PCI function in multiple Linux subsystems"

That is certainly true with Intel rdma / netdev use-case. We have a single PCI function owned by the netdev PCI driver that needs to be exposed in a different subsystem, i.e. rdma.

Reply:
I see. We need to then come up with a more generic description as to what type of device can attach to this bus.

Reply:
> @ranj063 Maybe clarify this refers to Linux MFD framework. PCI devices can also host multiple function and PCI multifunction devices are a different thing. This is about exposing a single PCI function (the audio DSP) in multiple Linux subsystems.

@kv2019i well its not that we're exposing it in multiple subsystems. Particularly in SOF, thats not the case. And the moment we say single-function, then the question will be why split a single-function at all